pgjdbc/src/main/java/com/yugabyte/ysql/LoadBalanceProperties.java

```diff
@@ -84,29 +79,28 @@ static void clearConnectionManagerMap() {
     }
   }
 
-  public static LoadBalanceProperties getLoadBalanceProperties(String url, Properties properties) {
-    LoadBalancerKey key = new LoadBalancerKey(url, properties);
+  public static LoadBalanceProperties getLoadBalanceProperties(LoadBalancerKey key) {
     LoadBalanceProperties lbp = loadBalancePropertiesMap.get(key);
     if (lbp == null) {
       synchronized (LoadBalanceProperties.class) {
         lbp = loadBalancePropertiesMap.get(key);
         if (lbp == null) {
-          lbp = new LoadBalanceProperties(url, properties);
+          lbp = new LoadBalanceProperties(key);
           loadBalancePropertiesMap.put(key, lbp);
         }
       }
     }
     return lbp;
   }
 
-  private LoadBalanceProperties(String origUrl, Properties origProperties) {
-    originalUrl = origUrl;
-    originalProperties = (Properties) origProperties.clone();
-    ybURL = processURLAndProperties();
+  private LoadBalanceProperties(LoadBalancerKey key) {
+    originalUrl = key.getUrl();
+    originalProperties = (Properties) key.getProperties().clone();
   }
 
-  public String processURLAndProperties() {
-    String[] urlParts = this.originalUrl.split("\\?");
+  public static ProcessedProperties processURLAndProperties(String url, Properties properties) {
```

Comment:
If the url is indeed getting parsed earlier in `Driver.parseURL()`, we do not have to repeat it here. We can simply read from the properties bag alone.